From 3d23274eefcf9c494358662539c29eebd8996cbf Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Mon, 23 Feb 2026 17:00:24 +0100
Subject: [PATCH 1/3] CMakeLists.txt: Fix CMAKE_CXX_FLAGS_{RELEASE,DEBUG} being
 lists of strings

g++ /* ... */ -O2 -Wall -DNDEBUG;-Wno-implicit-function-declaration -Wno-incompatible-pointer-types /* ... */ -o CMakeFiles/sdlnp2kai_sdl2.dir/sound/fmgen/fmgen_file.cpp.o -c /build/source/sound/fmgen/fmgen_file.cpp
ld: <glibc>/lib/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
bash: line 1: -Wno-implicit-function-declaration: command not found
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5a9884f..9c14d94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -323,8 +323,8 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 if(NOT EMSCRIPTEN)
 set(COMMON_C_CXX_FLAGS "-Wno-implicit-function-declaration -Wno-incompatible-pointer-types -Wno-incompatible-function-pointer-types -Wno-int-conversion -fno-sized-deallocation")
-set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -Wall -DDEBUG -DTRACE" ${COMMON_C_CXX_FLAGS})
-set(CMAKE_CXX_FLAGS_RELEASE "-O2 -Wall -DNDEBUG" ${COMMON_C_CXX_FLAGS})
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -Wall -DDEBUG -DTRACE ${COMMON_C_CXX_FLAGS}")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2 -Wall -DNDEBUG ${COMMON_C_CXX_FLAGS}")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "-Os -Wall")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-Og -g -Wall -DNDEBUG")
 endif()
-- 
2.52.0

